xrpl/models/utils.py

```diff
@@ -1,9 +1,44 @@
 """Helper util functions for the models module."""
+
+import re
 from dataclasses import dataclass, is_dataclass
-from typing import Any, Dict, List, Type, TypeVar, cast
+from typing import Any, Dict, List, Optional, Pattern, Type, TypeVar, cast
+
+from typing_extensions import Final
 
 from xrpl.models.exceptions import XRPLModelException
 
+# Note: This regex is used in DIDSet and Credential-related transactions. In addition to
+# this regex, Credential-related transactions requires the input strings to have
+# positive (non-zero) length
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]*")
+
+# Credentials are represented in hex. Whilst they are allowed a maximum length of 64
+# bytes, every byte requires 2 hex characters for representation
+_MAX_CREDENTIAL_LENGTH: int = 64 * 2
```

Comment:
```suggestion
_MAX_CREDENTIAL_LENGTH: Final[int] = 64 * 2
```

Reply:
My understanding is that `Final[]` is similar to the `const` keyword in cpp. Its an indication to type-checkers that this variable must not be written into, post-initialization. But if non-static elements of the program do write into the variable, type-checkers cannot detect such mistakes. Is that correct?